lms/djangoapps/discussion_api/api.py

```diff
@@ -276,7 +286,12 @@ def get_thread_list(request, course_key, page, page_size, topic_id_list=None, te
         "per_page": page_size,
         "text": text_search,
     }
+
     text_search_rewrite = None
+
+    if view:
+        query_params[view] = "true"
```

Comment:
An Http request is made. Via the url, it points to the correct view which puts the query parameters into a form and then validates the forms (checks for correct field values, in this case if the view is "unanswered" or "unread" else returns an error that says that an incorrect value has been given). Then it calls the api with the validated form data.